packages/flutter/lib/src/material/menu_style.dart

```diff
@@ -0,0 +1,447 @@
+  final MaterialStateProperty<double?>? elevation;
+
+  /// The padding between the menu's boundary and its child.
+  final MaterialStateProperty<EdgeInsetsGeometry?>? padding;
+
+  /// The minimum size of the menu itself.
+  ///
+  /// This value must be less than or equal to [maximumSize].
+  final MaterialStateProperty<Size?>? minimumSize;
+
+  /// The menu's size.
+  ///
+  /// This size is still constrained by the style's [minimumSize]
+  /// and [maximumSize]. Fixed size dimensions whose value is
+  /// [double.infinity] are ignored.
+  ///
+  /// To specify menus with a fixed width and the default height use
+  /// `fixedSize: Size.fromWidth(320)`. Similarly, to specify a fixed
+  /// height and the default width use `fixedSize: Size.fromHeight(100)`.
+  final MaterialStateProperty<Size?>? fixedSize;
+
+  /// The maximum size of the menu itself.
+  ///
+  /// A [Size.infinite] or null value for this property means that
+  /// the menu's maximum size is not constrained.
+  ///
+  /// This value must be greater than or equal to [minimumSize].
+  final MaterialStateProperty<Size?>? maximumSize;
+
+  /// The color and weight of the menu's outline.
+  ///
+  /// This value is combined with [shape] to create a shape decorated
+  /// with an outline.
+  final MaterialStateProperty<BorderSide?>? side;
+
+  /// The shape of the menu's underlying [Material].
+  ///
+  /// This shape is combined with [side] to create a shape decorated
+  /// with an outline.
+  final MaterialStateProperty<OutlinedBorder?>? shape;
+
+  /// The cursor for a mouse pointer when it enters or is hovering over
+  /// this menu's [InkWell].
+  final MaterialStateProperty<MouseCursor?>? mouseCursor;
+
+  /// Defines how compact the menu's layout will be.
+  ///
+  /// {@macro flutter.material.themedata.visualDensity}
+  ///
+  /// See also:
+  ///
+  ///  * [ThemeData.visualDensity], which specifies the [visualDensity] for all widgets
+  ///    within a [Theme].
+  final VisualDensity? visualDensity;
+
+  /// Determines the desired alignment of the submenu when opened relative to
+  /// the button that opens it.
+  ///
+  /// If there isn't sufficient space to open the menu with the given alignment,
+  /// and there's space on the other side of the button, then the alignment is
+  /// swapped to it's opposite (1 becomes -1, etc.), and the menu will try to
+  /// appear on the other side of the menu. If there isn't enough space there
+  /// either, then the menu will be pushed as far over as necessary to display
+  /// as much of itself as possible, possibly overlapping the parent button.
+  ///
+  /// Defaults to [AlignmentDirectional.topEnd], which means that in a
+  /// left-to-right context, the menu will appear with its origin at the upper
+  /// right corner of the [MenuButton] that owns it. This applies for vertical
+  /// menus (e.g. created by a [MenuButton] or [createMaterialMenu]).
+  ///
+  /// For a [MenuBar], the default is [AlignmentDirectional.bottomStart].
```

Comment:
We usually only document defaults in the component classes.

Reply:
Okay, I removed that bit.